### DISCUSSION TITLE
Recommendations for showing subscription plan pricing when a promo code is supplied

### DISCUSSION THREAD

**Discussion by robbyrussell** [pay-rails/pay#449] [Q&A] [answered]
Context: Recently inherited a Ruby on Rails project and it's using Pay for integration with Stripe. They're on v2.2.2 at the moment and we're exploring our upgrade options.

In parallel, we're trying to provide them with a quick solution to having promo codes that that not just give a user a discounted rate when they're billed BUT also show them the applied discount rate on subscription plans before they pull their credit card out.

Since the app isn't using Stripe's checkout widget, I'm curious if there is an existing pattern for showing plan pricing with a promo code (and/or the associated coupon's % off) without doing the math ourselves (but rather rely on Stripe showing the  discounted pricing).

Any pointers? It's my first time using this particular gem.

**Comment by excid3** [comment, marked as answer]
Sup @robbyrussell 👋 

You'd just include the promo code in the `subscribe` call. The additional options get passed over to the Stripe API. See: https://github.com/pay-rails/pay/blob/v2.2.2/lib/pay/stripe/billable.rb#L56

You'll probably need to add something to keep track of which promo code and the discount that's applied. We're not currently saving any of those details.

**Reply by robbyrussell** [reply]
Thanks, @excid3 that seems good for the transaction/purchase phase but was also thinking a step or two before that when we want to display a discounted price they would pay for. Am thinking we might need to do the calculation on our end to show what that would be.

**Reply by excid3** [reply]
Correct. We don't do any of that currently.

**Reply by rathboma** [reply]
@excid3 I'd like to sync 'discounts' back from Stripe into the Pay::Subscription model, I couldn't find a way to do that in the docs -- is there a way to add extra fields to the sync?

**Reply by jonmchan** [reply]
@rathboma I ended up just calling:

```Stripe::Subscription.retrieve(id: subscription.processor_id, expand: ['discounts'])```

**Reply by excid3** [reply]
Discounts are automatically synced: https://github.com/pay-rails/pay/blob/main/app/models/pay/stripe/subscription.rb#L132

**Reply by jonmchan** [reply]
Thanks for sharing that - I should have checked! Save me extra calls when querying this thing.
